src/Microsoft.DotNet.Wpf/src/PresentationFramework/System/Windows/Navigation/NavigationService.cs

```diff
@@ -2799,30 +2798,6 @@ private WebRequest CreateWebRequest(Uri resolvedDestinationUri, NavigateInfo nav
                 if (launched == LaunchResult.NotLaunched)
                     throw;
             }
-            catch (SecurityException e)
-            {
-                LaunchResult launched = LaunchResult.NotLaunched;
-
-                // the scenario this code is enabling is navigation to Uri's outside of the app
-                // for top-level.
-                // So for example at an express app at domain http://www.example.com
-                // click on a hyperlink to http://www.msn.com
-                // We will get a security exception on the attempt to access msn.
-                // So we delegate back to the top-level browser.
-                //
-                // IMPORTANT: Creating a WebRequest for a file:// URI doesn't fail here if the URI is outside
-                // the site of origin. Instead, WebRequest.EndGetResponse() will throw SecurityException.
-                // There is a similar case for such URIs there.
-                // Callers of this method should not assume that the application has access to the given URI.
-
-                if (e.PermissionType == typeof(System.Net.WebPermission))
-                {
-                    launched = AppSecurityManager.SafeLaunchBrowserOnlyIfPossible(CurrentSource, resolvedDestinationUri, IsTopLevelContainer);
-                }
-
-                if (launched == LaunchResult.NotLaunched)
-                    throw;
-            }
 
```

Comment:
Are you sure we don't need to still catch the exception and set `LaunchResult` to `NotLaunched` before rethrowing the exception?

Reply:
I don't think we need to set LaunchResult to anything since we now have full trust on WebPermission and the local variable is not useful anymore. I re-added catching the exception and just throw now